### ISSUE TITLE
[FEA] - AST Support for fixed_point literals.

### ISSUE THREAD

**Issue by breta-nv** [rapidsai/cudf#16118]
I wish I could use fixed_point literals in AST.

Our team utilizes compute_column, and we'd like to use fixed_point literals in the ast::expression. Without fixed_point, we have to rely on floating-point, which leads to incorrectly results due to the usual precision and rounding errors.

Without AST support, we have to rely on a slower fallback path.

**Comment by JayjeetAtGithub** [comment]
In #16088, I added support for `fixed_point_scalar` literals in AST [here](https://github.com/rapidsai/cudf/pull/16088/files#diff-19a4cb1509100c6de73610dd11fe902e651c3b2bd379e9244d2d5d7d0b39aa12), which allows me to now pass `fixed_point_scalar` in an `cudf::ast::operation`. Although, the code compiles now, I am getting some weird behaviour when I am applying such an expression to `cudf::compute_column`. 

So, basically, I have 2 filter expressions, each of them has a `fixed_point_scalar`. In one of them the value is `0.05` and in the other the value is `24`. I am representing both these values as a `fixed_point_scalar` with a defined scale of `-2` (same as my columns scale). The filter with `24` works correctly but the one with `0.05` does not. I am still trying to figure it out. But, I would like to know if anything is weird, or wrong that I am doing in this whole setup. Thanks a lot.

**Comment by kevkrist** [comment]
Any updates on this? This feels like an important feature that isn't getting any attention.

**Comment by vyasr** [comment]
No, we haven't gotten around to this yet. So far we've observed limited usage of our AST machinery with fixed point data, and we're also investing a lot right now in building out our JIT capabilities instead (see https://github.com/rapidsai/cudf/issues/15366) as a more flexible long-term solution, so this issue hasn't really been prioritized by the team.

**Comment by matal-nvidia** [comment]
Didn't [#20147](https://github.com/rapidsai/cudf/pull/20147) add support for decimal literals? Are we missing something?

**Comment by kevkrist** [comment]
Yes this is obsolete now I believe it can be closed. Fix #21980 is still not released though.

**Comment by davidwendt** [comment]
Yes, you are correct.
With #21980 and #22122 fixed-point decimal is supported as AST operands for add, subtract, multiply, divide and comparators like equal, greater, lesser.

**Comment by vyasr** [comment]
Whoops I totally forgot we merged that. Sorry about that.
